packages/vega-parser/src/parsers/guides/guide-util.js

```diff
@@ -86,3 +87,80 @@ export function extendOffset(value, offset) {
     : !isObject(value) ? { value, offset }
     : { ...value, offset: extendOffset(value.offset, offset) };
 }
+
+const DISCRETE_SCALES = new Set(['ordinal', 'band', 'point']);
+
+function formatType(_, scaleType) {
+  const formatType = _('formatType');
+
+  if (formatType) {
+    return formatType;
+  }
+  if (scaleType === Time) {
+    return 'time'
+  }
+  if (scaleType === UTC) {
+    return 'utc'
+  }
+}
+
+function wrapFormat(text, format, formatType) {
+  if (format == null) {
+    return text
+  }
+  const func = formatType === "time" ? "timeFormat" : formatType === "utc" ? "utcFormat" : "format";
+  return `${func}(${text}, "${format}")`;
+}
+
+function domainText(scaleType, scaleName, format, formatType) {
+  return DISCRETE_SCALES.has(scaleType)
+    ? ` with values " + domain('${scaleName}')`
+    : ` from " + ${wrapFormat(
+        `domain("${scaleName}")[0]`,
+        format,
+        formatType
+      )} + " to " + ${wrapFormat(
+        `domain("${scaleName}")[1]`,
+        format,
+        formatType
+      )}`;
+}
+
+function titleText(_) {
+  let title = _('title')
+
+  if (isArray(title)) {
+    // only use the first row of multiline titles
+    title = title[0];
+  }
+
+  if (title) {
+    return isObject(title) ? ` showing " + ${title.signal} + "` : ` showing ${title}`;
+  }
+
+  return '';
+}
+
+export function legendAriaLabel(_, scope) {
+  const domains = [];
+  for (const legendType of ['fill', 'stroke', 'opacity', 'size', 'shape', 'strokeDash']) {
+    const scaleName = _(legendType);
+    if (scaleName) {
+      const scaleType = scope.scales[scaleName].params.type;
+      const domain = domainText(scaleType, scaleName, _('format'), formatType(_, scaleType));
+      domains.push(`${domain} + " as ${legendType}`);
+    }
+  }
+  const signal = `"legend${titleText(_)}${domains.join(',')}"`;
+  return { signal };
+}
+
+export function axisAriaLabel(_, scope) {
+  const scaleName = _('scale');
+  const scaleType = scope.scales[scaleName].params.type;
+  const orient = _('orient')
+  const axisType = orient === 'bottom' || orient === 'top' ? 'x' : 'y';
```

Comment:
Again, please use defined constants in constants.js

Reply:
Done